### PR TITLE
Don't leave unused nodes in administrativeMetadata

### LINKED ISSUE #2644: Repeated edits to an APO create a lot of XML nodes
**Describe the bug**
Repeatedly updating an APO (using https://github.com/sul-dlss/argo/pull/2529) is adding a lot of XML nodes to the administrativeMetadata. 

The administrativeMetadata for https://argo-qa.stanford.edu/view/druid:vr838wn4827 looks like this after many rounds of changed:

```xml
<?xml version="1.0"?>
<administrativeMetadata>
  <dissemination>
    
  </dissemination>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination/>
  <registration/>
  <dissemination>
    <workflow id=""/>
  </dissemination>
  <registration>
    <workflow id="accessionWF"/>
    <workflow id="goobiWF"/>
    <workflow id="registrationWF"/>
    <collection id="druid:ns375cy3379"/>
  </registration>
</administrativeMetadata>
```

**User Impact**
It's kind of a mess but doesn't seem to hurt functionality.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #2644

### CODE REVIEW

File: app/services/cocina/to_fedora/administrative_metadata.rb [hunk 1]
```diff
@@ -6,19 +6,50 @@ module ToFedora
...
       end
+
+      def clear_nodes
+        admin_node.xpath('registration/workflow|registration/collection|dissemination/workflow').each(&:remove)
+      end
+
+      def add_dissemination_nodes
+        return unless administrative.disseminationWorkflow
+
+        dissemination_node = admin_node.xpath('dissemination').first || admin_node.add_child('<dissemination/>').first
```
Comment (justinlittman) [inline app/services/cocina/to_fedora/administrative_metadata.rb:33]: If nodes are cleared, how can there be a first?

File: app/services/cocina/to_fedora/administrative_metadata.rb [hunk 2]
```diff
@@ -6,19 +6,50 @@ module ToFedora
...
+        dissemination_node.add_child "<workflow id=\"#{administrative.disseminationWorkflow}\" />"
+      end
+
+      def add_registration_nodes
+        registration_node = admin_node.xpath('registration').first || admin_node.add_child('<registration/>').first
```
Comment (justinlittman) [inline app/services/cocina/to_fedora/administrative_metadata.rb:38]: If nodes are cleared, how can there be a first?

File: app/services/cocina/to_fedora/administrative_metadata.rb [hunk 3]
```diff
@@ -6,19 +6,50 @@ module ToFedora
...
+        Array(administrative.registrationWorkflow).each do |wf_id|
+          registration_node.add_child "<workflow id=\"#{wf_id}\" />"
+        end
+        Array(administrative.collectionsForRegistration).each do |collection|
+          registration_node.add_child "<collection id=\"#{collection}\" />"
+        end
+      end
+
+      def admin_node
+        @admin_node ||= begin
```
Comment (justinlittman) [inline app/services/cocina/to_fedora/administrative_metadata.rb:48]: Trivial, but this could be a one liner instead of a four liner.

**Review by justinlittman — CHANGES_REQUESTED** [review]
Marked request changes for questions.

File: app/services/cocina/to_fedora/administrative_metadata.rb (hunk 1 above)
Comment (jcoyne) [reply app/services/cocina/to_fedora/administrative_metadata.rb:33]: The child nodes are cleared, not this one

File: app/services/cocina/to_fedora/administrative_metadata.rb (hunk 2 above)
Comment (jcoyne) [reply app/services/cocina/to_fedora/administrative_metadata.rb:38]: Same answer as above

File: app/services/cocina/to_fedora/administrative_metadata.rb (hunk 3 above)
Comment (jcoyne) [reply app/services/cocina/to_fedora/administrative_metadata.rb:48]: Can you explain what you would like to see here?

Comment (justinlittman) [reply app/services/cocina/to_fedora/administrative_metadata.rb:48]: `@admin_node ||= administrative_metadata.ng_xml.xpath('//administrativeMetadata').first`

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of suggestion.

File: app/services/cocina/to_fedora/administrative_metadata.rb (hunk 3 above)
Comment (jcoyne) [reply app/services/cocina/to_fedora/administrative_metadata.rb]: Sounds good. Change made.

**Pull request merged**